### PR TITLE
chore(deps): upgrade Firebase to v9

### PR DESCRIPTION
**Pull request by thisislawatts**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Developer experience (improves developer workflows for contributing to the project)

## Description

Upgrades firebase 

- chore(deps): upgrade to firebase@9
- chore(cypress): install firebase dependency
- refactor: firebase switch to modular usage for auth

Following recommended process: https://firebase.google.com/docs/web/modular-upgrade

Additional Issues will be created to track the following process:
* [x] ~~Upgrade auth to drop use of `compat` layer~~
* [ ] Upgrade firestore to drop use of `compat` layer
* [ ] Upgrade functions to drop use of `compat` layer
* [ ] Upgrade database to drop use of `compat` layer
* [ ] Upgrade storage to drop use of `compat` layer

### CODE REVIEW

File: src/stores/User/user.store.test.tsx [hunk 1]
```diff
@@ -1,32 +1,51 @@
...
+//         },
+//       }),
+//     currentUser: {
+//       displayName: 'testDisplayName',
+//       uid: 'testUid',
+//     },
+//     onAuthStateChanged: jest.fn(),
+//     signInWithEmailAndPassword: jest.fn(),
+//   },
+// }))
```
Comment (evakill) [inline src/stores/User/user.store.test.tsx:48]: remove?

File: src/stores/User/user.store.ts [hunk 2]
```diff
@@ -32,7 +40,7 @@ export class UserStore extends ModuleStore {
   public user: IUserPPDB | null | undefined
 
   @observable
-  public authUser: firebase.default.User | null
+  public authUser: any | null // TODO: Fix type
```
Comment (evakill) [inline src/stores/User/user.store.ts:43]: fix?

**Review by evakill — COMMENTED** [review]
testing locally....

**Comment by evakill** [comment]
pulled and ran locally - all worked for me!
not sure if this is new from this change - i'm getting a warning on function test (no failures tho)
```
 /Users/evakillenberg/community-platform-official/functions/node_modules/firebase-admin/lib/auth/auth-api-request.js:913
>              return Promise.reject(new error_1.FirebaseAuthError(error_1.AuthClientErrorCode.INVALID_EMAIL));
>                                    ^
>  
>  FirebaseAuthError: The email address is improperly formatted.
>      at AuthRequestHandler.getAccountInfoByEmail (/Users/evakillenberg/community-platform-official/functions/node_modules/firebase-admin/lib/auth/auth-api-request.js:913:35)
>      at Auth.getUserByEmail (/Users/evakillenberg/community-platform-official/functions/node_modules/firebase-admin/lib/auth/base-auth.js:156:40)
>      at /Users/evakillenberg/community-platform-official/functions/dist/index.js:2830:44
>      at Generator.next (<anonymous>)
>      at /Users/evakillenberg/community-platform-official/functions/dist/index.js:2792:71
>      at new Promise (<anonymous>)
>      at __webpack_modules__.1328.__awaiter (/Users/evakillenberg/community-platform-official/functions/dist/index.js:2788:12)
>      at isValidEmailCreationRequest (/Users/evakillenberg/community-platform-official/functions/dist/index.js:2829:65)
>      at /Users/evakillenberg/community-platform-official/functions/dist/index.js:2256:71
>      at Generator.next (<anonymous>) {
>    errorInfo: {
>      code: 'auth/invalid-email',
>      message: 'The email address is improperly formatted.'
>    },
>    codePrefix: 'auth'
>  }
```

**Review by benfurber — APPROVED** [review]
Lgtm.

One question, should `validationSchema` on `src/pages/SignUp/SignUp.tsx` also pull from `FRIENDLY_MESSAGES`?

File: src/stores/User/user.store.ts (hunk 2 above)
Comment (thisislawatts) [reply src/stores/User/user.store.ts]: Good catch, and fixed it.

**Pull request merged**